sdk/communication/communication-job-router/src/routerClient.ts

```diff
@@ -0,0 +1,673 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/// <reference lib="esnext.asynciterable" />
+
+import { logger } from "./models/logger";
+import { SDK_VERSION } from "./constants";
+import { createPipelineFromOptions, InternalPipelineOptions, RestResponse } from "@azure/core-http";
+import "@azure/core-paging";
+import { PagedAsyncIterableIterator } from "@azure/core-paging";
+import {
+  UpsertDistributionPolicyOptions,
+  UpsertExceptionPolicyOptions,
+  CreateJobOptions,
+  DeleteChannelOptions,
+  DeleteDistributionPolicyOptions,
+  DeleteExceptionPolicyOptions,
+  CancelJobOptions,
+  AcceptJobOptions,
+  DeclineJobOptions,
+  CompleteJobOptions,
+  DeleteQueueOptions,
+  DeregisterWorkerOptions,
+  GetChannelOptions,
+  GetClassificationPolicyOptions,
+  GetDistributionPolicyOptions,
+  GetExceptionPolicyOptions,
+  GetJobOptions,
+  GetQueueOptions,
+  GetWorkerOptions,
+  ListChannelsOptions,
+  ListClassificationPoliciesOptions,
+  ListDistributionPoliciesOptions,
+  ListExceptionPoliciesOptions,
+  ListJobsOptions,
+  ListEnqueuedJobsOptions,
+  ListQueuesOptions,
+  ListWorkersOptions,
+  RegisterWorkerOptions,
+  RouterClientOptions,
+  UpsertQueueOptions,
+  CloseJobOptions,
+  ReleaseWorkerOptions,
+  DeleteClassificationPolicyOptions,
+  UpdateJobClassificationOptions,
+  UpdateJobLabelsOptions,
+  GetJobPositionOptions,
+  UpsertChannelOptions,
+  UpsertClassificationPolicyOptions
+} from "./models/options";
+import { ChannelType } from "./models/models";
+import {
+  ClassificationPolicy,
+  UpsertClassificationPolicyResponse,
+  CreateJobResponse,
+  UpsertQueueResponse,
+  DistributionPolicy,
+  ExceptionPolicy,
+  JobRouterApiClient,
+  UpsertChannelRequest,
+  UpsertChannelResponse,
+  UpsertClassificationPolicyRequest,
+  UpsertExceptionPolicyResponse,
+  UpsertDistributionPolicyResponse,
+  UpsertDistributionPolicyRequest,
+  UpsertExceptionPolicyRequest,
+  JobPositionDetails,
+  AcceptJobOfferResponse,
+  JobQueue,
+  RouterChannel,
+  RouterJob,
+  RouterWorker,
+  JobStateSelector,
+  WorkerStateSelector
+} from "./generated/src";
+
+import { KeyCredential, TokenCredential } from "@azure/core-auth";
+import {
+  createCommunicationAuthPolicy,
+  isKeyCredential,
+  parseClientArguments
+} from "@azure/communication-common";
+import { createSetHeadersPolicy } from "./policies";
+import {
+  LabelValue,
+  RestCreateJobRequest,
+  RestReclassifyJobRequest,
+  RestRegisterWorkerRequest,
+  RestUpsertQueueRequest
+} from "./models/requests";
+
+/**
+ * Checks whether the type of a value is RouterClientOptions or not.
```

Comment:
```suggestion
 * Checks whether the type of a value is {@link RouterClientOptions} or not.
```

Since code documentation is intended to be rendered, we like using proper markdown and proper tsdoc references 🙂